crates/starknet-devnet-types/src/patricia_key.rs

```diff
@@ -75,6 +77,26 @@ impl TryFrom<PatriciaKey> for starknet_api::state::StorageKey {
     }
 }
 
+impl TryFrom<FieldElement> for PatriciaKey {
+    type Error = Error; // Replace with the appropriate error type
+
+    fn try_from(element: FieldElement) -> Result<Self, Self::Error> {
+        let bytes = Felt::new(element.to_bytes_be())?; // Replace with the appropriate method to convert FieldElement to bytes
+        PatriciaKey::new(bytes) // Replace with the appropriate error handling
+    }
+}
+
+impl TryFrom<Result<FieldElement, NonAsciiNameError>> for PatriciaKey {
+    type Error = Error;
+
+    fn try_from(value: Result<FieldElement, NonAsciiNameError>) -> Result<Self, Self::Error> {
+        match value {
+            Ok(field_element) => PatriciaKey::try_from(field_element),
+            Err(_) => Err(Error::ProgramError),
+        }
+    }
+}
```

Comment:
Is this an idiomatic thing to do? To implement `TryFrom` on a `Result` type?

Reply:
I think is not the best way. I can remove this TryFrom for the result type but then i would need to use 2 unwraps on the same line, i.e:

let storage_var_address = starknet_types::patricia_key::PatriciaKey::try_from(
            get_storage_var_address("ERC20_balances", &[FieldElement::from(self.account_address)]).unwrap(),
        )
        .unwrap();

what do you think?

Reply:
Please your advice @FabijanC @marioiordanov @ivpavici

Reply:
Sorry for waiting, we'll re-review this week

Reply:
It will be better to implement try_from on NonResult type. Can't be 100% sure that every time the Result will have the following generic params: FieldElement, NonAsciiNameError

Reply:
Done @marioiordanov @FabijanC @ivpavici , please check again in PR #405 (by mistake i closed this PR)